notification/service/notification.go

```diff
@@ -122,8 +154,8 @@ func (s *notificationServiceImpl) send(ctx context.Context, c *client.Client, ms
 }
 
 // createClientWithContextSigner creates with a signer based on current context
-func (s *notificationServiceImpl) createClientWithContextSigner(ctx context.Context) (*client.Client, error) {
-	c, err := s.createClient()
+func (s *notificationServiceImpl) createClientWithContextSigner(ctx context.Context, options ...configuration.HTTPClientOption) (*client.Client, error) {
```

Comment:
could we have a single `createClient` method that would also have a `WithContextSigner` option to do what's on lines 162-167 ?  Something like:
````
func WithContextSigner(ctx context.Context) configuration.HttpClientOption {
  return func(c *client.Client) {
    sgn := signer.NewSATokenSigner(ctx)
    saTokenSigner, err := sgn.Signer()
    if err != nil {
 		return err
    }
    c.SetJWTSigner(saTokenSigner)
    return nil
  }
}
````
(with a change in `configuration.HttpClientOption` to return an error)